File: src/rollup.py
import pandas as pd
import datetime as dt
import pathlib

# Determine the project root directory
project_root = pathlib.Path(__file__).parent.parent
data_dir = project_root / "data"

today = dt.date.today()
yesterday = today - dt.timedelta(days=1)

live_file = data_dir / "live.dta.csv"
rollup_file = data_dir / f"{yesterday.isoformat()}.csv"

if live_file.exists():
    df = pd.read_csv(live_file, delimiter=";")
    df["DateTime"] = pd.to_datetime(df["DateTime"], format="%d.%m.%Y %H:%M:%S")
    
    yesterday_df = df[df["DateTime"].dt.date == yesterday]
    
    if not yesterday_df.empty:
        yesterday_df.to_csv(rollup_file, index=False, sep=";")
        
        # Remove the rolled-up data from the live file
        remaining_df = df[df["DateTime"].dt.date != yesterday]
        remaining_df.to_csv(live_file, index=False, sep=";")

File: src/rollup.js
const fs = require('fs');
const path = require('path');
const Papa = require('papaparse');

const projectRoot = path.resolve(__dirname, '..');
const dataDir = path.join(projectRoot, 'data');

const liveFile = path.join(dataDir, 'live.dta.csv');
const today = new Date();
const yesterday = new Date(today);
yesterday.setDate(yesterday.getDate() - 1);
const yesterdayISO = yesterday.toISOString().split('T')[0];
const rollupFile = path.join(dataDir, `${yesterdayISO}.csv`);

if (fs.existsSync(liveFile)) {
    const csvData = fs.readFileSync(liveFile, 'utf-8');
    const { data: rows, meta } = Papa.parse(csvData, { header: true, skipEmptyLines: true });

    const yesterdayRows = [];
    const remainingRows = [];

    rows.forEach(row => {
        // Assuming DateTime is in a format that can be parsed by new Date()
        // e.g., "14.07.2025, 13:40:00"
        const [datePart, timePart] = row.DateTime.split(', ');
        const [day, month, year] = datePart.split('.');
        const formattedDate = `${year}-${month}-${day}T${timePart}`;
        const rowDate = new Date(formattedDate);

        if (rowDate.toISOString().startsWith(yesterdayISO)) {
            yesterdayRows.push(row);
        } else {
            remainingRows.push(row);
        }
    });

    if (yesterdayRows.length > 0) {
        const rollupCsv = Papa.unparse(yesterdayRows, { header: true, delimiter: ';', newline: '\r\n' });
        fs.writeFileSync(rollupFile, rollupCsv);
        console.log(`Created rollup file: ${rollupFile}`);
    }

    if (remainingRows.length > 0) {
        const remainingCsv = Papa.unparse(remainingRows, { header: true, delimiter: ';', newline: '\r\n' });
        fs.writeFileSync(liveFile, remainingCsv);
    } else {
        // If all rows were from yesterday, the live file becomes empty
        fs.unlinkSync(liveFile);
    }
}
